src/shared/store/photoSlice.ts

```diff
@@ -0,0 +1,96 @@
+import {createAsyncThunk, createSlice} from '@reduxjs/toolkit';
+
+import API from '../../core/services/API';
+import {IPhoto} from '../interfaces';
+import {IPhotoState} from '../interfaces/StateSlices';
+import {isPending, isRejected} from './helpers';
+
+
+export const addPhoto = createAsyncThunk(
+  'photo/addPhoto',
+  async function (photo: IPhoto, thunkAPI) {
+    try {
+      return await API.post('/api/photos', photo) as IPhoto;
+    } catch (error) {
+      return thunkAPI.rejectWithValue(error);
+    }
+  }
+);
+
+export const updatePhoto = createAsyncThunk(
+  'photo/updatePhoto',
+  async function (photo: Required<IPhoto>, thunkAPI) {
+    try {
+      return await API.patch('/api/photos', photo) as IPhoto;
+    } catch (error) {
+      return thunkAPI.rejectWithValue(error);
+    }
+  }
+);
+
+export const getPhoto = createAsyncThunk(
+  'photo/getPhoto',
+  async function (id: number[], thunkAPI) {
+    try {
+      return await API.get(`/api/photos${id === null ? `?ids=${[...id]}` : ''}`) as IPhoto[] | IPhoto;
```

Comment:
Can you do it `Photo[]` if there are multiple `id`s and 0 `id`s, and `IPhoto` if there is a single `id`? Same for Album